src/webgl/ShaderGenerator.js

```diff
@@ -1638,3 +1638,503 @@ export default shadergenerator;
 if (typeof p5 !== 'undefined') {
   p5.registerAddon(shadergenerator)
 }
+
+
+
+/* ------------------------------------------------------------- */
+/**
+ * @method getWorldInputs
+ * @description
+ * Registers a callback to modify the world-space properties of each vertex in a shader. This hook can be used inside <a href="#/p5/baseColorShader">baseColorShader()</a>.modify() and similar shader modify calls to customize vertex positions, normals, texture coordinates, and colors before rendering. "World space" refers to the coordinate system of the 3D scene, before any camera or projection transformations are applied.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseNormalShader">baseNormalShader()</a>
+ * - <a href="#/p5/baseColorShader">baseColorShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {function} callback
+ *        A callback function which receives a vertex object containing position (vec3), normal (vec3), texCoord (vec2), and color (vec4) properties. The function should return the modified vertex object.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseMaterialShader().modify(() => {
+ *     let t = uniformFloat(() => millis());
+ *     getWorldInputs(inputs => {
+ *       // Move the vertex up and down in a wave in world space
+ *       // In world space, moving the object (e.g., with translate()) will affect these coordinates
+ *       inputs.position.y += 0.5 * sin(t * 0.001 + inputs.position.x * 0.05);
```

Comment:
I think maybe the wrong one got updated -- a scale of 20 makes sense here because in world space, the sphere is 50 units tall. The update to change the scale to 0.5 was for *object space* where the sphere is 2 units tall.
```suggestion
 *       inputs.position.y += 20 * sin(t * 0.001 + inputs.position.x * 0.05);
```

Reply:
Changed the wave displacement scale from `0.5` to `20` for the world-space example, matching the object’s actual size.